src/Temporalio/Worker/Tuning/ISlotSupplier.cs

```diff
@@ -6,10 +6,13 @@ namespace Temporalio.Worker.Tuning
     /// Slot suppliers control how slots are handed out for workflow and activity tasks as well as
     /// local activities when used in conjunction with a <see cref="WorkerTuner"/>.
     ///
-    /// Currently you cannot implement your own slot supplier, but you can use the provided <see
-    /// cref="FixedSizeSlotSupplier"/> and <see cref="ResourceBasedSlotSupplier"/> slot suppliers.
+    /// Pre-built slot suppliers are available as
+    /// <see cref="FixedSizeSlotSupplier"/> and <see cref="ResourceBasedSlotSupplier"/>.
+    ///
+    /// In order to implement your own slot supplier, you can implement the
+    /// <see cref="ICustomSlotSupplier"/> interface.
     /// </summary>
     public interface ISlotSupplier
```

Comment:
If you do change `ICustomSlotSupplier` to an abstract class, this could also be an abstract class with an `internal` constructor thereby making it where people _can't_ accidentally provide an invalid instance.

Reply:
This one doesn't work so much because the built-in impls are records and thus can't extend from a class, only interfaces. 

We could change them to classes, but that's technically breaking, though that's fine at this point.

Reply:
:+1: I don't have a strong opinion. I'm ok leaving this as an interface or doing the compat break on the fixed/resource suppliers.